breaking: remove param files in folder in favor of `params.js/ts` file (#16189)
sveltejs/kit: documentation/docs/30-advanced/10-advanced-routing.md

## Matching

A route like `src/routes/fruits/[page]` would match `/fruits/apple`, but it would also match `/fruits/rocketship`. We don't want that. You can ensure that route parameters are well-formed by adding a _matcher_ to your `src/params.js` file (or `src/params.ts`)...

```js
/// file: src/params.js
import { defineParams } from '@sveltejs/kit';

export const params = defineParams({
	fruit: (param) => {

> 		if (param !== 'apple' && param !== 'orange') throw new Error('Invalid fruit');

Reviewer: ```suggestion
		if (param !== 'apple' && param !== 'orange') return;
```

src/routes/fruits/[page+++=fruit+++]
```

If the pathname doesn't match, SvelteKit will try to match other routes (using the sort order specified below), before eventually returning a 404. If it does match, the returned value is passed as the param value.

You can also use a [Standard Schema](https://standardschema.dev) — for example with [Valibot](https://valibot.dev):

```js
/// file: src/params.js
import { defineParams } from '@sveltejs/kit';
import * as v from 'valibot';

export const params = defineParams({
	number: v.pipe(v.string(), v.toNumber())
});
```

> When a schema is used, SvelteKit validates the parameter and uses the transformed output as the param value. If validation fails, the route does not match.

Reviewer: ```suggestion
If validation fails, the route does not match. If it succeeds, the param is typed with the output type of the schema, which must extend `string`, `boolean`, `number` (as in the example above) or `bigint`:
```

```js
/// file: src/routes/items/[id=number]/+page.js
/** @type {import('./$types').PageLoad} */
export function load({ params }) {
	console.log(typeof params.id); // 'number'
}
```

> Matchers must return a `string`, `boolean`, `number` or `BigInt` value. That way `resolve` can require the correct type and easily stringify them.

Reviewer: ```suggestion
Transformations should be _symmetrical_ — if converting to a number, then calling `toString()` on that number should return the original string. This allows the [`resolve`]($app-paths#resolve) function to construct a pathname correctly:
```
